bed/bedPe/bedPe.go

```diff
@@ -0,0 +1,185 @@
+// Package bedPe provides functions for handling Bed Paired End format files, used for
+// handling pairs of genomic regions.
+
+package bedPe
+
+import (
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"io"
+	"log"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+type BedPe struct {
+	Chrom1            string
+	ChromStart1       int
+	ChromEnd1         int
+	Chrom2            string
+	ChromStart2       int
+	ChromEnd2         int
+	Name              string
+	Score             int
+	Strand1           bed.Strand
+	Strand2           bed.Strand
+	FieldsInitialized int      //number of fields that are initialized, used for smart writing.
+	Annotation        []string //long form for extra fields
+}
+
+// String converts a BedPe struct to a string so it will be automatically formatted when printing with the fmt package.
+func (b BedPe) String() string {
+	return ToString(b, b.FieldsInitialized)
+}
+
+// ToString converts a BedPe struct into a BedPe file format string. Useful for writing to files or printing.
+func ToString(bunk BedPe, fields int) string {
+	switch {
+	case fields == 6:
+		return fmt.Sprintf("%s\t%d\t%d\t%s\t%d\t%d", bunk.Chrom1, bunk.ChromStart1, bunk.ChromEnd1, bunk.Chrom2, bunk.ChromStart2, bunk.ChromEnd2)
+	case fields == 7:
+		return fmt.Sprintf("%s\t%d\t%d\t%s\t%d\t%d\t%s", bunk.Chrom1, bunk.ChromStart1, bunk.ChromEnd1, bunk.Chrom2, bunk.ChromStart2, bunk.ChromEnd2, bunk.Name)
+	case fields == 8:
+		return fmt.Sprintf("%s\t%d\t%d\t%s\t%d\t%d\t%s\t%d", bunk.Chrom1, bunk.ChromStart1, bunk.ChromEnd1, bunk.Chrom2, bunk.ChromStart2, bunk.ChromEnd2, bunk.Name, bunk.Score)
+	case fields == 9:
+		return fmt.Sprintf("%s\t%d\t%d\t%s\t%d\t%d\t%s\t%d\t%c", bunk.Chrom1, bunk.ChromStart1, bunk.ChromEnd1, bunk.Chrom2, bunk.ChromStart2, bunk.ChromEnd2, bunk.Name, bunk.Score, bunk.Strand1)
+	case fields == 10:
+		return fmt.Sprintf("%s\t%d\t%d\t%s\t%d\t%d\t%s\t%d\t%c\t%c", bunk.Chrom1, bunk.ChromStart1, bunk.ChromEnd1, bunk.Chrom2, bunk.ChromStart2, bunk.ChromEnd2, bunk.Name, bunk.Score, bunk.Strand1, bunk.Strand2)
+	case fields >= 11:
+		var out string = fmt.Sprintf("%s\t%d\t%d\t%s\t%d\t%d\t%s\t%d\t%c\t%c", bunk.Chrom1, bunk.ChromStart1, bunk.ChromEnd1, bunk.Chrom2, bunk.ChromStart2, bunk.ChromEnd2, bunk.Name, bunk.Score, bunk.Strand1, bunk.Strand2)
+		for i := range bunk.Annotation {
+			out = fmt.Sprintf("%s\t%s", out, bunk.Annotation[i])
+		}
+		return out
+	default:
+		log.Fatalf("Error: expecting a request to print at least 6 bedPe fields, but got: %d\n", fields)
+	}
+	return ""
+}
+
+// WriteBed writes an input BedPe struct to an io.Writer.
+func WriteBedPe(file io.Writer, input BedPe) {
```

Comment:
I am having trouble telling the difference between WriteBedPe and WriteToFileHandle.  If these are the same, I would keep the second name.